src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useCallback } from 'react';
+
+import { EuiButton, EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import { useEffect } from 'react';
+import { CoreStart, WorkspaceAttribute } from '../../../../../core/public';
+
+type WorkspaceOption = EuiComboBoxOptionOption<WorkspaceAttribute>;
+
+interface WorkspaceDropdownListProps {
+  coreStart: CoreStart;
+  onCreateWorkspace: () => void;
+  onSwitchWorkspace: (workspaceId: string) => void;
+}
+
+function workspaceToOption(workspace: WorkspaceAttribute): WorkspaceOption {
+  return { label: workspace.name, key: workspace.id, value: workspace };
+}
+
+export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
+  const { coreStart, onCreateWorkspace, onSwitchWorkspace } = props;
+  const [loading, setLoading] = useState(true);
+  const [currentWorkspace, setCurrentWorkspace] = useState([] as WorkspaceOption[]);
+  const [workspaceOptions, setWorkspaceOptions] = useState([] as WorkspaceOption[]);
+
+  const onSearchChange = useCallback(
+    async (searchValue: string) => {
+      setLoading(true);
+      const response = await coreStart.workspaces.client.list({
```

Comment:
What about add an observable workspace list in the public workspace client? So that the workspace list will be updated after workspace created, updated and deleted.

Reply:
I think it makes sense to use Observable currentWorkspace and Workspace list